### PR TITLE
Add unit test for empty password

### PR DESCRIPTION
**Pull request by loay**
### Description
Passwords with empty strings should be rejected
Forwardport of https://github.com/strongloop/loopback/pull/3481
prerequisite: https://github.com/strongloop/loopback-datasource-juggler/pull/1435

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

### CODE REVIEW

**Comment by loay** [comment]
@slnode test please

**Comment by loay** [comment]
@slnode test please

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ cache:
   directories:
     - travis_phantomjs
 before_install:
+  - npm config set registry http://ci.strongloop.com:4873/
```
Comment (ssh24) [inline .travis.yml]: Can we do this on a different PR so we can revert/port whenever needed?

Comment (loay) [inline .travis.yml]: Same issue of cross-dependency ..
A different pr will not pass, as it needs  the content of the first commit .. The first commit will not pass without the 2nd of a juggler release ...
I didn't intend to squash anyways .. 2 different commits for 2 different purposes!

Comment (ssh24) [inline .travis.yml]: No I meant a different PR for just setting up the travis config:
```
 before_install:
   - npm config set registry http://ci.strongloop.com:4873/
```

If you are not squashing then this also works.
